message/tests/test_job.py

```diff
@@ -21,101 +20,132 @@ class TestMessage(unittest.TestCase):
+        populated_msg, populated_dict = self._populated()
+        serialized = populated_msg.serialize()
+        actual = populated_msg.deserialize(serialized)
+        self.assertEqual(actual, populated_dict)
 
     def test_populate_from_dict_overwrite_true(self):
-        """ Overwritten from a dictionary values"""
-        actual = self.empty_msg
-        actual.populate(self.populated_dict, overwrite=True)
-        self.assertEqual(attr.asdict(actual), self.populated_dict)
+        """ Overwritten from a dictionary values """
+        _, populated_dict = self._populated()
+        actual, _ = self._empty()
+        actual.populate(populated_dict, overwrite=True)
+        self.assertEqual(attr.asdict(actual), populated_dict)
 
     def test_populate_from_dict_overwrite_false(self):
         """ Do not overwrite from dictionary """
-        self.populated_dict['job_id'] = 123
-        self.populated_dict['rb_number'] = 33333
-        actual = copy.deepcopy(self.populated_msg)
-        actual.populate(self.populated_dict, overwrite=False)
-        self.assertEqual(actual.rb_number, self.populated_msg.rb_number)
+        populated_msg, populated_dict = self._populated()
+        populated_dict['job_id'] = 123
+        populated_dict['rb_number'] = 33333
+        actual, _ = self._populated()
+        actual.populate(populated_dict, overwrite=False)
+        self.assertEqual(actual.rb_number, populated_msg.rb_number)
         self.assertEqual(actual.job_id, 123)
 
     def test_populate_from_serialized_overwrite_true(self):
         """ Overwrite with a serialized object """
-        serialized = self.populated_msg.serialize()
-        actual = self.empty_msg
+        populated_msg, populated_dict = self._populated()
+        serialized = populated_msg.serialize()
+        actual, _ = self._empty()
         actual.populate(source=serialized, overwrite=True)
-        self.assertEqual(attr.asdict(actual), self.populated_dict)
+        self.assertEqual(attr.asdict(actual), populated_dict)
 
     def test_populate_from_serialized_overwrite_false(self):
         """ Do not overwrite from a serialized object """
-        new_msg = copy.deepcopy(self.populated_msg)
+        new_msg, _ = self._populated()
         new_msg.job_id = 123
         new_msg.rb_number = 33333
         serialized = new_msg.serialize()
-        actual = copy.deepcopy(self.populated_msg)
+        actual, _ = self._populated()
+        original, _ = self._populated()
         actual.populate(source=serialized, overwrite=False)
-        self.assertEqual(actual.rb_number, self.populated_msg.rb_number)
+        self.assertEqual(actual.rb_number, original.rb_number)
         self.assertEqual(actual.job_id, 123)
 
     def test_invalid_serialized(self):
         """ Test raise exception for invalid serialized object """
         serialized = 'test'
-        self.assertRaises(ValueError, self.empty_msg.populate, serialized)
+        empty_msg, _ = self._populated()
```

Comment:
Should empty_msg be assigned from _empty()?
i.e. `empty_msg, _ = self._empty()` 

(Not that this affects the test much, since whether or not the message is populated vs empty shouldn't have an effect on this specific test)